### PR TITLE
IDEX-3356: Fix remote calls updating and removing ssh keys

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java [hunk 1]
```diff
@@ -281,7 +285,7 @@ private String sshKeyAttributeName(String host, int isPrivate) {
     @SuppressWarnings("unchecked")
     ProfileDescriptor getUserProfile() throws ApiException, IOException, JsonParseException {
         final String profileDescriptor = HttpJsonHelper.requestString(profileApiUrl, "GET", null);
-        return JsonHelper.fromJson(profileDescriptor, ProfileDescriptor.class, new TypeToken<ProfileDescriptor>() {}.getType());
+        return DtoFactory.getInstance().createDtoFromJson(profileDescriptor, ProfileDescriptor.class);
```
Comment (skabashnyuk) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: DtoFactory.newDto?

Comment (akorneta) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: here DtoFactory.getInstance().createDtoFromJson is ok

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java [hunk 2]
```diff
@@ -280,21 +284,29 @@ private String sshKeyAttributeName(String host, int isPrivate) {
...
-                                     JsonHelper.toJson(preferences),
-                                     Pair.of("Content-Type", "application/json"));
+        HttpJsonHelper.request(profileApiUrl + "/prefs",
+                               HttpMethod.POST,
+                               new JsonStringMapImpl<>(preferences),
+                               Pair.of("Content-Type", "application/json"));
+    }
+
+    @SuppressWarnings("unchecked")
+    void removeSshKey(List<String> keys) throws ApiException, IOException {
```
Comment (garagatyi) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: This method removes list of keys. Change method name in accordance to that

Comment (skabashnyuk) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: +

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java [hunk 3]
```diff
@@ -280,21 +284,29 @@ private String sshKeyAttributeName(String host, int isPrivate) {
 
     @SuppressWarnings("unchecked")
     ProfileDescriptor getUserProfile() throws ApiException, IOException, JsonParseException {
-        final String profileDescriptor = HttpJsonHelper.requestString(profileApiUrl, "GET", null);
-        return JsonHelper.fromJson(profileDescriptor, ProfileDescriptor.class, new TypeToken<ProfileDescriptor>() {}.getType());
+        final String profileDescriptor = HttpJsonHelper.requestString(profileApiUrl, HttpMethod.GET, null);
```
Comment (garagatyi) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: HttpJsonHelper can parse response to DTO object. We don't need manual parsing here.

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java [hunk 4]
```diff
@@ -280,21 +284,29 @@ private String sshKeyAttributeName(String host, int isPrivate) {
...
+                               new JsonStringMapImpl<>(preferences),
+                               Pair.of("Content-Type", "application/json"));
+    }
+
+    @SuppressWarnings("unchecked")
+    void removeSshKey(List<String> keys) throws ApiException, IOException {
+        HttpJsonHelper.request(profileApiUrl + "/prefs",
+                               HttpMethod.DELETE,
+                               new JsonArrayImpl<>(keys),
+                               Pair.of("Content-Type", "application/json"));
```
Comment (garagatyi) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: AFAIK it should be array of query params, not headers. BTW HTTPJsonHelper sends content type header already.

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java (hunk 2 above)
Comment (akorneta) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: ok

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java (hunk 4 above)
Comment (akorneta) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: :+1:

File: plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java (hunk 3 above)
Comment (akorneta) [inline plugin-ssh/che-plugin-ssh-ext-sshkey/src/main/java/org/eclipse/che/ide/ext/ssh/server/UserProfileSshKeyStore.java]: :+1:

**Comment by garagatyi** [comment]
LGTM

**Comment by skabashnyuk** [comment]
LGTM

**Pull request merged**